### PR TITLE
Fix country change in checkout form

### PR DESCRIPTION
**Pull request by martinfojtik**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR will fix the bug after a country change in checkout form.
| Type?         | bug fix 
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-2227
| How to test?  | Follow steps in BOOM-2227

### CODE REVIEW

**Comment by vincentbz** [comment]
Hi @martinfojtik 

Do I have to create a new shop with that PR ? because when I change the country in the address, I don't see the fields anymore, see gif below.

Thanks !

adresse change country

**Comment by martinfojtik** [comment]
Hi @vincentbz ,

you don't need to create a new shop. There was a bug in my code. Now it should work.

**Comment by vincentbz** [comment]
Hi @martinfojtik 

I still have the same issue on your PR :-/

File: controllers/front/AddressController.php [hunk 1]
```diff
@@ -129,4 +129,26 @@ public function getBreadcrumbLinks()
 
         return $breadcrumb;
     }
+
+    public function displayAjaxAddressForm()
+    {
+        $addressForm = $this->makeAddressForm();
+
+        if (Tools::getIsset('id_address')) {
+            $addressForm->loadAddressById(Tools::getValue('id_address'));
```
Comment (kpodemski) [inline controllers/front/AddressController.php]: `(int`)?

File: controllers/front/AddressController.php [hunk 2]
```diff
@@ -129,4 +129,26 @@ public function getBreadcrumbLinks()
...
+        }
+
+        if (Tools::getIsset('id_country')) {
+            $addressForm->fillWith(array('id_country' => Tools::getValue('id_country')));
```
Comment (kpodemski) [inline controllers/front/AddressController.php]: `(int)`?

**Comment by Quetzacoalt91** [comment]
Hello @martinfojtik,

I was working on it yesterday by looking at a duplicated issue. I'm glad to see you already worked on it. :smiley: 

However, because I see all deliveries steps see may change with the ajax test, and an new ajax request is implemented although there is already an existing one, I tried to minimize your changes to bring the same fix.

I got something by reusing the template inheritance, this allow us to reuse the address ajax update and just modify the URLs. You can see the changes on #7710.

**Comment by maximebiloe** [comment]
Hello @martinfojtik,

As we merged the #7710, I close this one.
Is will be shipped into the `1.7.1.x`.

Regards

**Pull request closed without merging**